### ISSUE TITLE
bug(evals): PrecisionRecallFScore macro/weighted F-score diverges from scikit-learn `average` semantics

### ISSUE THREAD

**Issue by kimnamu** [Arize-ai/phoenix#13739]
First, thank you for `phoenix-evals` — the code evaluators are a clean, dependency-light way to compute classification metrics, and `PrecisionRecallFScore` is exactly the kind of thing that's nice to have without pulling in scikit-learn.

I think I may have found a small discrepancy in the `macro` and `weighted` F-score, and I'd like to confirm it before proposing a fix.

## Summary

`PrecisionRecallFScore` computes the `macro`/`weighted` F-beta from the **already-aggregated** precision and recall, rather than computing F-beta **per class and then averaging**. scikit-learn (which the class's docstring mirrors via its `average` parameter) does the latter. The two differ whenever per-class F is not a linear function of the aggregated P/R (i.e. essentially always for multi-class).

Precision, recall, and **micro** F are unaffected (for micro, precision == recall, so F from the aggregate is already correct). The **binary `positive_label`** path is also unaffected.

## Where

`packages/phoenix-evals/src/phoenix/evals/metrics/precision_recall.py`

In `_evaluate` the F-score is always computed from the aggregated P/R:

```python
precision, recall = self._aggregate_precision_recall(counts_by_label)
suffix = "" if self.average == "macro" else f"_{self.average}"

f_score = self._compute_f_score(precision, recall, self.beta)  # uses aggregated P, R
```

`_aggregate_precision_recall` already loops per class to build P and R, but discards per-class F.

## Reproduction (no API key / network needed — pure computation)

```python
from phoenix.evals.metrics import PrecisionRecallFScore

expected = ["cat", "dog", "cat", "bird"]
output   = ["cat", "cat", "cat", "bird"]

for avg in ("macro", "weighted"):
    s = {x.name: x.score for x in PrecisionRecallFScore(average=avg).evaluate(
        {"expected": expected, "output": output})}
    print(avg, s)
```

Per class: cat P=2/3 R=1 F1=0.8, dog P=0 R=0 F1=0, bird P=1 R=1 F1=1.

| average | metric | phoenix (current) | scikit-learn 1.9.0 | P / R (both tools) |
|---------|--------|-------------------|--------------------|--------------------|
| macro    | F1 | 0.606061 (= 60/99) | **0.600000** | P=0.555556, R=0.666667 ✅ match |
| weighted | F1 | 0.656250 (= 21/32) | **0.650000** | P=0.583333, R=0.750000 ✅ match |
| micro    | F1 | 0.750000           | 0.750000     | P=R=0.75 ✅ match (no divergence) |

scikit-learn reference:

```python
from sklearn.metrics import precision_recall_fscore_support as prfs
prfs(expected, output, average="macro")     # -> (0.5556, 0.6667, 0.6000, None)
prfs(expected, output, average="weighted")  # -> (0.5833, 0.7500, 0.6500, None)
```

The gap is larger when a class has undefined precision under a custom `zero_division`, and it also shows up for `beta != 1`.

## Proposed fix (small)

Compute per-class F-beta inside the existing `_aggregate_precision_recall` loop and average it with the **same** strategy already used for precision/recall (macro = arithmetic mean, weighted = support-weighted mean). `micro` and the binary `positive_label` path stay exactly as they are. I have a minimal patch (~30 lines, plus tests aligning the existing assertions to the scikit-learn values) ready if this looks right to you.

Per CONTRIBUTING I'm filing this first and won't open the PR until a maintainer confirms it's worth pursuing. Happy to be corrected if I'm misreading the intended semantics of `average` here.

---
*This report was prepared with the help of an AI agent (Claude Code); a human reviewed the reproduction, the scikit-learn comparison, and the proposed fix before posting.*
